middleware/configuration/source/model/transform.cpp

```diff
@@ -398,13 +398,20 @@ namespace casual
 
                   if( source.groups)
                   {
-                     result.groups = common::algorithm::transform( source.groups.value(), []( auto& group)
+                     auto defaults = [ &source]() -> std::string
+                     {
+                        if( source.defaults && source.defaults->directory)
+                           return *source.defaults->directory;
+                        return {};
+                     }();
+
+                     result.groups = common::algorithm::transform( source.groups.value(), [ &defaults]( auto& group)
                      {
                         queue::Group result;
 
                         result.alias = group.alias.value_or( "");
                         result.note = group.note.value_or( "");
-                        result.queuebase = group.queuebase.value_or( "");
+                        result.queuebase = group.queuebase.value_or( defaults != "" ? defaults + "/" + result.alias + ".qb" : "");
```

Comment:
I'm not sure that _configuration_ should be responsible for the queuebase name. At least the decision should be made in one place.
I think the _queue module_ should have the responsibility. If so, we need to extend the model.queue.group with _directory_, and make the decision when configure the queue-group with the model.